db/indices.c

```diff
@@ -345,27 +527,50 @@ int upd_record_indices(struct ireq *iq, void *trans, int *opfailcode,
                        blob_buffer_t *del_idx_blobs, int same_genid_with_upd,
                        unsigned long long vgenid, int *deferredAdd)
 {
+    int rc = 0;
     char *od_dta_tail = NULL;
     int od_tail_len;
     char *od_olddta_tail = NULL;
     int od_oldtail_len;
 
+    void *cur = NULL;
+    dtikey_t delditk = {0}; // will serve as the delete key obj
+    dtikey_t ditk = {0};    // will serve as the add or upd key obj
+    bool reorder = gbl_reorder_idx_writes && iq->usedb->sc_from != iq->usedb &&
```

Comment:
so why sc and constraints?

Reply:
I think i saw issues while schema changing. For now lets not reorder if schemachanging and if this table has constraints.